frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -1064,6 +1115,40 @@ const sinkURIDataModel = (
   knDataModel.edges.push(...getEventTopologyEdgeItems(resource, resources.brokers));
 };
 
+export const createEventSinkTopologyNodeData = (
+  resource: K8sResourceKind,
+  overviewItem: OverviewItem,
+  type: string,
+  defaultIcon: string,
+  operatorBackedService: boolean = false,
+): TopologyDataObject => {
+  const { monitoringAlerts = [] } = overviewItem;
+  const dcUID = _.get(resource, 'metadata.uid');
+  const deploymentsLabels = _.get(resource, 'metadata.labels', {});
+  const deploymentsAnnotations = _.get(resource, 'metadata.annotations', {});
+
+  const builderImageIcon =
+    getImageForIconClass(`icon-${deploymentsLabels['app.openshift.io/runtime']}`) ||
+    getImageForIconClass(`icon-${deploymentsLabels['app.kubernetes.io/name']}`);
+  return {
+    id: dcUID,
+    name: resource?.metadata.name || deploymentsLabels['app.kubernetes.io/instance'],
+    type,
+    resource,
+    resources: { ...overviewItem, isOperatorBackedService: operatorBackedService },
+    data: {
+      monitoringAlerts,
+      kind: referenceFor(resource),
+      editURL: deploymentsAnnotations['app.openshift.io/edit-url'],
+      vcsURI: deploymentsAnnotations['app.openshift.io/vcs-uri'],
+      vcsRef: deploymentsAnnotations['app.openshift.io/vcs-ref'],
+      builderImage: builderImageIcon || defaultIcon,
+      isKnativeResource: type === NodeType.EventSink,
+      kameletType: KameletType.Sink,
+    },
+  };
+};
```

Comment:
Look like we are passing unwanted properties to data. Eg. `monitoringAlerts`, `editURL`, `vcsURI`, `vcsRef` and `builderImage` as we are using `getEventSourceIcon` to get the icon for the node. We can remove these from the data. Wdyt?

```suggestion
export const createEventSinkTopologyNodeData = (
  resource: K8sResourceKind,
  overviewItem: OverviewItem,
  type: string,
  defaultIcon: string,
  operatorBackedService: boolean = false,
): TopologyDataObject => {
  const dcUID = _.get(resource, 'metadata.uid');
  return {
    id: dcUID,
    name: resource?.metadata.name,
    type,
    resource,
    resources: { ...overviewItem, isOperatorBackedService: operatorBackedService },
    data: {
      kind: referenceFor(resource),
      isKnativeResource: type === NodeType.EventSink,
      kameletType: KameletType.Sink,
    },
  };
};
```